### ISSUE TITLE
DocService should provide a way to navigate endpoint request objects without leaving debug form.

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#43]
Currently, if one of the parameters in the request is a struct type, clicking on it will navigate away from the debug page. This makes it difficult to input a debug request while looking through the documentation. It would be nice if you could stay on the debug page while navigating structs / substructs. Perhaps it could open on a frame on the right-side of the page.

**Comment by trustin** [comment]
@delegacy, interested?

**Comment by delegacy** [comment]
When a user clicks the link for structs on the debug page with the desktop version, show the documentation in a modal view. What do you think?

**Comment by anuraaga** [comment]
A modal would be better than the current behavior, but I think it'd be much better to be able to edit the form while viewing the definition. That's why a frame feels better to me.

**Comment by trustin** [comment]
Hover detail panel?

**Comment by trustin** [comment]
or.. maybe something like this: https://github.com/kjantzer/backbonejs-panel-view so that a user can navigate through the nested structs.

**Comment by trustin** [comment]
Demo: http://kjantzer.github.io/backbonejs-panel-view/

**Comment by anuraaga** [comment]
The panel looks good, though it creates a modal, so unless there's a better supported way in the API it might need these css hacks

```css
.panel-overlay {
  display: none;
}

#_panel {
  pointer-events: none;
}
```

**Comment by trustin** [comment]
Didn't mean to use it as it is. Perhaps we could write something real simple rather than adding another dependency? :-)

**Comment by anuraaga** [comment]
Ah yeah - UI-wise I like it :)

**Comment by delegacy** [comment]
Yes, The way using hovering has weakness when navigating nested structs. I like your idea of a dedicated collapsible area for those documentation.

**Comment by minwoox** [comment]
I think this was already implemented, so let me close this issue.
